### ISSUE TITLE
WAF rules

### ISSUE THREAD

**Issue by netsysadm** [hap-wi/roxy-wi#294]
I been disabling WAF rules, but even if I unchecked(then restart WAF service) the WAF rules is still enabled.. any recommendations what to remediate this please?

**Comment by Aidaho12** [comment]
Check the logs

**Comment by netsysadm** [comment]
it is working by the way, the logs shows that it is on, I've had it as on detection, because when I tried it to On, it will block any http access due the rules HTTP Policy on the WAF rules. here's the log, not sure what thats attribute error means 
![image](https://user-images.githubusercontent.com/56064755/135750379-3efb0a35-b16c-4531-82e9-0e837484e6b5.png)

**Comment by Aidaho12** [comment]
It seems a bug. I will fix it in the next release

**Comment by netsysadm** [comment]
Thank you will wait for your update!

**Comment by netsysadm** [comment]
Hi @Aidaho12 

Is it possible to ship the WAF logs to a syslog? I can only see from the GUI to send logs of haproxy.log

another issue I notice is that the dashboard for the WAF is only Null and not currently generating anything or is it a feature running for the paid feature, SMON, metrics, etc?

![image](https://user-images.githubusercontent.com/56064755/135822979-f6713f85-892b-42b3-9b3e-a979f66285ad.png)

**Comment by Aidaho12** [comment]
It ship to syslog. the file is /var/log/sec... log

Yes. this graf generated by metrics service

**Comment by netsysadm** [comment]
the option is only for Haproxy.log on the GUI, may I know where can I change for the modsec_auditlog as well?

![image](https://user-images.githubusercontent.com/56064755/135824650-c390e1b7-9024-4548-9cc6-4cf7b49520e7.png)

**Comment by Aidaho12** [comment]
this option means that Roxy-WI will go to the syslog server for logs, but not reconfigure services

**Comment by netsysadm** [comment]
Thanks @Aidaho12 

what if I was blocked by the WAF(as I was simulating a scan/DDOS) where can I remove the blocking of that specific server IP address

**Comment by Aidaho12** [comment]
Somewhere in waf

**Comment by netsysadm** [comment]
in the WAF gui I can only see the on/off/restart buttons, logs and the rules/options. or is it on the commandline? thank you and really apologies for too many questions. lol

**Comment by Aidaho12** [comment]
Yes, via cli

**Comment by netsysadm** [comment]
i did look into the 2 haproxy server and the roxy-wi server but I was able to find only the waf.conf and some other config files, any recommendation?

**Comment by Aidaho12** [comment]
Read about modSec :)

Could you send me part the log file, where is this error appears?

**Comment by netsysadm** [comment]
its on the WAF logs page

**Comment by Aidaho12** [comment]
I understand it, send me this logs, which are displayed

**Comment by netsysadm** [comment]
--4f57142e-H--
Message: Access denied with code 500 (phase 1) (Error: Connection drop not implemented on this platform. Operator EQ matched 1 at IP:dos_block. [file "/etc/haproxy/waf/rules/modsecurity_crs_11_dos_protection.conf"] [line "16"] [id "981045"]
Message: Internal error: REQUEST_BODY phase incomplete for input filter in phase 1
Action: Intercepted (phase 1)
Apache-Handler: IIS
Stopwatch: 1633399662000586 586653 (- - -)
Stopwatch2: 1633399662000586 586653; combined=212, p1=106, p2=0, p3=0, p4=0, p5=70, sr=33, sw=36, l=0, gc=0
Producer: ModSecurity for nginx (STABLE)/2.9.2 (http://www.modsecurity.org/); OWASP_CRS/2.2.9.
Server: ModSecurity Standalone
Engine-Mode: "ENABLED"

--4f57142e-Z--

--e5c6e50c-A--
[05/Oct/2021:10:10:35 +0800]  127.0.0.1 80 127.0.0.1 80
--e5c6e50c-B--
GET /login.php 1.1
user-agent: Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0
accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
accept-language: en-US,en;q=0.5
referer: https://www.test.local/index.php
cookie: PHPSESSID=36uag6ocp84di6ou7s14vrs1to; security=low
upgrade-insecure-requests: 1
cache-control: max-age=0
host: www.test.local

--e5c6e50c-H--
Message: Access denied with code 500 (phase 1) (Error: Connection drop not implemented on this platform. Operator EQ matched 0 at IP. [file "/etc/haproxy/waf/rules/modsecurity_crs_11_dos_protection.conf"] [line "11"] [id "981044"] [msg "Denial of Service (DoS) Attack Identified from 127.0.0.1 (65 hits since last alert)"]
Action: Intercepted (phase 1)
Apache-Handler: IIS
Stopwatch: 1633399835000092 93413 (- - -)
Stopwatch2: 1633399835000092 93413; combined=363, p1=272, p2=0, p3=0, p4=0, p5=61, sr=190, sw=30, l=0, gc=0
Producer: ModSecurity for nginx (STABLE)/2.9.2 (http://www.modsecurity.org/); OWASP_CRS/2.2.9.
Server: ModSecurity Standalone
Engine-Mode: "ENABLED"

--e5c6e50c-Z--

--234e2a50-A--
[05/Oct/2021:10:10:41 +0800]  127.0.0.1 80 127.0.0.1 80
--234e2a50-B--
POST /login.php 1.1
user-agent: Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0
accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
accept-language: en-US,en;q=0.5
content-type: application/x-www-form-urlencoded
content-length: 88
origin: https://www.test.local
referer: https://www.test.local/login.php
cookie: PHPSESSID=36uag6ocp84di6ou7s14vrs1to; security=low
upgrade-insecure-requests: 1
host: www.test.local

--234e2a50-C--
username=admin&password=password&Login=Login&user_token=ccad73193dbe57f37ae5126092a54ae1
--234e2a50-H--
Message: Access denied with code 500 (phase 1) (Error: Connection drop not implemented on this platform. Operator EQ matched 1 at IP:dos_block. [file "/etc/haproxy/waf/rules/modsecurity_crs_11_dos_protection.conf"] [line "16"] [id "981045"]
Message: Internal error: REQUEST_BODY phase incomplete for input filter in phase 1
Action: Intercepted (phase 1)
Apache-Handler: IIS
Stopwatch: 1633399841000618 618811 (- - -)
Stopwatch2: 1633399841000618 618811; combined=276, p1=162, p2=0, p3=0, p4=0, p5=77, sr=58, sw=37, l=0, gc=0
Producer: ModSecurity for nginx (STABLE)/2.9.2 (http://www.modsecurity.org/); OWASP_CRS/2.2.9.
Server: ModSecurity Standalone
Engine-Mode: "ENABLED"

--234e2a50-Z--

--c23dff7d-A--
[05/Oct/2021:10:10:44 +0800]  127.0.0.1 80 127.0.0.1 80
--c23dff7d-B--
POST /login.php 1.1
user-agent: Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0
accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
accept-language: en-US,en;q=0.5
referer: https://www.test.local/login.php
content-type: application/x-www-form-urlencoded
content-length: 88
origin: https://www.test.local
cookie: PHPSESSID=36uag6ocp84di6ou7s14vrs1to; security=low
upgrade-insecure-requests: 1
cache-control: max-age=0
host: www.test.local

--c23dff7d-C--
username=admin&password=password&Login=Login&user_token=ccad73193dbe57f37ae5126092a54ae1
--c23dff7d-H--
Message: Access denied with code 500 (phase 1) (Error: Connection drop not implemented on this platform. Operator EQ matched 1 at IP:dos_block. [file "/etc/haproxy/waf/rules/modsecurity_crs_11_dos_protection.conf"] [line "16"] [id "981045"]
Message: Internal error: REQUEST_BODY phase incomplete for input filter in phase 1
Action: Intercepted (phase 1)
Apache-Handler: IIS
Stopwatch: 1633399844000288 288376 (- - -)
Stopwatch2: 1633399844000288 288376; combined=241, p1=134, p2=0, p3=0, p4=0, p5=72, sr=46, sw=35, l=0, gc=0
Producer: ModSecurity for nginx (STABLE)/2.9.2 (http://www.modsecurity.org/); OWASP_CRS/2.2.9.
Server: ModSecurity Standalone
Engine-Mode: "ENABLED"

--c23dff7d-Z--

--0f841227-A--
[05/Oct/2021:10:15:23 +0800]  127.0.0.1 80 127.0.0.1 80
--0f841227-B--
POST /login.php 1.1
user-agent: Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0
accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
accept-language: en-US,en;q=0.5
referer: https://www.test.local/login.php
content-type: application/x-www-form-urlencoded
content-length: 88
origin: https://www.test.local
cookie: PHPSESSID=36uag6ocp84di6ou7s14vrs1to; security=low
upgrade-insecure-requests: 1
cache-control: max-age=0
host: www.test.local

--0f841227-C--
username=admin&password=password&Login=Login&user_token=ccad73193dbe57f37ae5126092a54ae1
--0f841227-H--
Message: Access denied with code 500 (phase 1) (Error: Connection drop not implemented on this platform. Operator EQ matched 0 at IP. [file "/etc/haproxy/waf/rules/modsecurity_crs_11_dos_protection.conf"] [line "11"] [id "981044"] [msg "Denial of Service (DoS) Attack Identified from 127.0.0.1 (21 hits since last alert)"]
Message: Internal error: REQUEST_BODY phase incomplete for input filter in phase 1
Action: Intercepted (phase 1)
Apache-Handler: IIS
Stopwatch: 1633400123000618 619800 (- - -)
Stopwatch2: 1633400123000618 619800; combined=499, p1=354, p2=0, p3=0, p4=0, p5=94, sr=237, sw=51, l=0, gc=0
Producer: ModSecurity for nginx (STABLE)/2.9.2 (http://www.modsecurity.org/); OWASP_CRS/2.2.9.
Server: ModSecurity Standalone
Engine-Mode: "ENABLED"

--0f841227-Z--

--fb515a5f-A--
[05/Oct/2021:10:15:35 +0800]  127.0.0.1 80 127.0.0.1 80
--fb515a5f-B--
POST /login.php 1.1
user-agent: Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0
accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
accept-language: en-US,en;q=0.5
content-type: application/x-www-form-urlencoded
content-length: 88
origin: https://www.test.local
referer: https://www.test.local/login.php
cookie: PHPSESSID=36uag6ocp84di6ou7s14vrs1to; security=low
upgrade-insecure-requests: 1
host: www.test.local

--fb515a5f-C--
username=admin&password=password&Login=Login&user_token=7bdd8e80891686c7e1ab196a0e8f630e
--fb515a5f-H--
Message: Access denied with code 500 (phase 1) (Error: Connection drop not implemented on this platform. Operator EQ matched 1 at IP:dos_block. [file "/etc/haproxy/waf/rules/modsecurity_crs_11_dos_protection.conf"] [line "16"] [id "981045"]
Message: Internal error: REQUEST_BODY phase incomplete for input filter in phase 1
Action: Intercepted (phase 1)
Apache-Handler: IIS
Stopwatch: 1633400135000475 475661 (- - -)
Stopwatch2: 1633400135000475 475661; combined=195, p1=114, p2=0, p3=0, p4=0, p5=54, sr=41, sw=27, l=0, gc=0
Producer: ModSecurity for nginx (STABLE)/2.9.2 (http://www.modsecurity.org/); OWASP_CRS/2.2.9.
Server: ModSecurity Standalone
Engine-Mode: "ENABLED"

--fb515a5f-Z--

**Comment by Aidaho12** [comment]
Thank you

**Comment by netsysadm** [comment]
I also encountered something. I've set the haproxy to log, but it still doesn't write to the haproxy.log, i've tried several other logging parameter but still to no avail. any recommendations again sir? Thank you!

below is my haproxy log config
![image](https://user-images.githubusercontent.com/56064755/135959302-a4ba3a48-d793-427d-af86-08c46b85a441.png)

![image](https://user-images.githubusercontent.com/56064755/135961320-0d0ca7fd-2bf9-455b-9441-4c9d0f456c1a.png)

**Comment by Aidaho12** [comment]
copy the rsyslog config from this repository on your haproxy server and restart it

btw, the bug has been fixed

**Comment by netsysadm** [comment]
hi created a rsyslog/haproxy.conf and copied the one I find here in your repo

```
if $programname startswith 'haproxy' then /var/log/haproxy.log
& stop

```
but still not working though..

**Comment by Aidaho12** [comment]
Have you restarted syslog?

**Comment by netsysadm** [comment]
yes I did

**Comment by Aidaho12** [comment]
Have you configured syslog work through the network?

**Comment by netsysadm** [comment]
yes as I'm still testing it to generate logs to local server(not to a remote server yet)

**Comment by Aidaho12** [comment]
then google it, probably you have made mistake in rsyslog config

**Comment by netsysadm** [comment]
Hi @Aidaho12 i was able to run the logging for the haproxy. but I've noticed that even if the haproxy logged in the the /var/log/haproxy.log

![image](https://user-images.githubusercontent.com/56064755/136151819-1451ce83-2d01-459e-bf07-de29205953af.png)

it's not showing on the GUI logs, or atleast its blank(when I checked the include WAF logs, then it will only show the logs that from the modsecurity-audit.logs

![image](https://user-images.githubusercontent.com/56064755/136151970-b35240b7-fc94-42e5-9b98-5dd7669262ab.png)
